src/Suggestor/Suggestor.ts

```diff
@@ -378,6 +407,53 @@ function addRecurrenceSuggestions(
     return results;
 }
 
+/*
+ * If the cursor is located in a section that is followed by a Blocked By Symbol, suggest options
+ * for what to enter as Depend on Option.
+ * It should contain suggestion of Possible Dependant Tasks
+ * of what the user is typing.
+ */
+function addBlockedBySuggestions(
+    line: string,
+    cursorPos: number,
+    settings: Settings,
+    blockedBySymbol: string,
+    allTasks: Task[],
+    taskToSuggestFor?: Task,
+) {
+    const results: SuggestInfo[] = [];
+
+    const blockedByRegex = new RegExp(`(${blockedBySymbol})([0-9a-zA-Z ^,]*,)*([0-9a-zA-Z ^,]*)`, 'ug');
+    const blockedByMatch = matchByPosition(line, blockedByRegex, cursorPos);
+    if (blockedByMatch && blockedByMatch.length >= 1) {
+        // blockedByMatch[1] = Blocked By Symbol
+        const existingBlockedByIdStrings = blockedByMatch[2] || '';
+        const newTaskToAppend = blockedByMatch[3].trim();
+
+        if (newTaskToAppend.length >= settings.autoSuggestMinMatch) {
+            const genericMatches = searchForCandidateTasksForDependency(
+                newTaskToAppend,
+                allTasks,
+                taskToSuggestFor,
+                [] as Task[],
```

Comment:
Should Probably add these Fields.

Reply:
> Should Probably add these Fields.

I don't understand that comment...

Reply:
> > Should Probably add these Fields.
> 
> I don't understand that comment...

The Empty Arrays there, should be filled.

They sort out the Task, if the Task has a Mention, Depending on the Task Being Edited.
(I just noticed, this doesn't Allow First Level Circular Dependencies, so this will Probably Stay, Empty or get removed?)

They sort out all tasks, which the Task being SuggestedFor already Depends On. (We don't need Duplicate Suggestions)

Reply:
It would be really good idea to not allow the user to create direct circular dependencies as it will create pairs of tasks that will never be found by the ‘not blocked’ search instruction, in the current implementation.

Reply:
I'm Aware of that Limitation.
In my understanding, even N Circular Dependencies are not shown with the Filter.
![image](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/16257607/e65421f2-deb3-495f-bc8f-94099eb6ba36)
Query Used:
```
tasks
path includes {{query.file.path}}
is not blocked
```

That's Exactly the Reason why I want to add a Graph View, to the Plugin. 
The Problem could Additionally be solved by displaying a warning somewhere that Circular Dependencies Exist.
Or Update one of the Filters to Contain the Circular Dependencies (With a warning.)

(For myself this is Exactly the use Case i want from Dependencies, Identifying Huge Circular Dependencies to get Started Somewhere.
Because i have So Many Tasks and Areas in my Live and many of the Interweave , its Hard too Keep an Overview and Identifie such Bottlenecks. As i Feel like im Often Stuck in Deadlocks and i wont even Notice.)

How do you want to Proceed with the Problem?
Not Allowing direct Dependencies doesn't solve the Problem, in my Opinion.

Reply:
> How do you want to Proceed with the Problem?
> Not Allowing direct Dependencies doesn't solve the Problem, in my Opinion.

I feel that this topic is really outside the scope of this PR...

As I understand it, the PR is using the same code as the Edit Task Modal to work out which tasks to suggest, which is fine for now.

Having said that, I don't have enough understanding of the detailed code changes in this PR right now...

So rather than thinking about cyclic dependencies, I'm going to merge the PR and then try to add unit tests for all the new behaviours, as the priority - so that this great new functionality can be released to users.

I tried testing it manually but there are too many combinations of behaviours to be tested, it's going to be more effective to try and record the behaviours in tests....

These were the combinations I was manually testing...

- Settings:
    - Minimum characters to type: 0, 1, 3
    - Number of suggestions to show: 6, 12
- Tasks format:
    - Emoji format
    - Dataview format
- Behaviours:
    - Which of the two options the user selects for ID - with or without a machine-generated ID


Resolving ....